components/searchable_channel_list.jsx

```diff
@@ -48,6 +48,12 @@ export default class SearchableChannelList extends React.Component {
         }
     }
 
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.isSearch) {
```

Comment:
Can you change this to `nextProps.isSearch && !this.props.isSearch` so that this only updates the state when you first start searching? Otherwise, it'll reset the page to 0 whenever the MoreChannels modal re-renders